services/firebase/resource.js

```diff
@@ -0,0 +1,6 @@
+import { getDatabase, ref, set } from 'firebase/database'
+
+export async function writeUserData(userInfo) {
+  const db = getDatabase()
+  return await set(ref(db, 'users/' + Date.now()), userInfo)
+}
```

Comment:
I'm not 100% sure but I think this should be an API route in NextJS. So it should be located in the api folder. I had a look at a similar example:https://github.com/leerob/nextjs-vercel-firebase 

If we set it up as an API route, we can do the `initialzeApp(fireBaseConfig)` call also there as we don't need it in our index page.